tests/integration/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -0,0 +1,835 @@
+                                                     settings_);
+  auto request = olp::dataservice::read::PartitionsRequest().WithFetchOption(
+      FetchOptions::CacheWithUpdate);
+  {
+    VolatileLayerClient::PartitionsResponse partitions_response;
+    olp::client::Condition condition;
+    client.GetPartitions(request,
+                         [&](VolatileLayerClient::PartitionsResponse response) {
+                           partitions_response = std::move(response);
+                           condition.Notify();
+                         });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+
+    // Request 1 return. Cached value (nothing)
+    ASSERT_FALSE(partitions_response.IsSuccessful())
+        << ApiErrorToString(partitions_response.GetError());
+  }
+
+  {
+    // Request 2 to check there is a cached value.
+    wait_for_end_signal->get_future().get();
+    request.WithFetchOption(CacheOnly);
+    VolatileLayerClient::PartitionsResponse partitions_response;
+    olp::client::Condition condition;
+    client.GetPartitions(request,
+                         [&](VolatileLayerClient::PartitionsResponse response) {
+                           partitions_response = std::move(response);
+                           condition.Notify();
+                         });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+    // Cache should be available here.
+    ASSERT_TRUE(partitions_response.IsSuccessful())
+        << ApiErrorToString(partitions_response.GetError());
+  }
+}
+
+TEST_F(VolatileLayerClientReadTest, GetPartitions403CacheClear) {
+  olp::client::HRN hrn(GetTestCatalog());
+  olp::dataservice::read::VolatileLayerClient client(hrn, "testlayer",
+                                                     settings_);
+  {
+    testing::InSequence s;
+    EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_PARTITIONS), _, _, _, _))
+        .Times(1);
+    EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_PARTITIONS), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(
+            olp::http::NetworkResponse().WithStatus(403), HTTP_RESPONSE_403));
+  }
+
+  // Populate cache
+  auto request = olp::dataservice::read::PartitionsRequest();
+  VolatileLayerClient::PartitionsResponse partitions_response;
+  olp::client::Condition condition;
+  client.GetPartitions(request,
+                       [&](VolatileLayerClient::PartitionsResponse response) {
+                         partitions_response = std::move(response);
+                         condition.Notify();
+                       });
+
+  ASSERT_TRUE(condition.Wait(kTimeout));
+  ASSERT_TRUE(partitions_response.IsSuccessful());
+
+  // Receive 403
+  request.WithFetchOption(OnlineOnly);
+  client.GetPartitions(request,
+                       [&](VolatileLayerClient::PartitionsResponse response) {
+                         partitions_response = std::move(response);
+                         condition.Notify();
+                       });
+  ASSERT_TRUE(condition.Wait());
```

Comment:
```suggestion
  ASSERT_TRUE(condition.Wait(kWaitTimeout));
```
Is it intended to be called without explicit timeout and relying on the default parameter?